pkg/sfu/buffer/helpers.go

```diff
@@ -351,4 +353,28 @@ func IsAV1Keyframe(payload []byte) bool {
 	}
 }
 
+// IsVP9Keyframe detects if vp9 payload is a keyframe
+// taken from https://github.com/jech/galene/blob/master/codecs/codecs.go
+// all credits belongs to Juliusz Chroboczek @jech and the awesome Galene SFU
+func IsVP9Keyframe(payload []byte) bool {
+	var vp9 codecs.VP9Packet
+	_, err := vp9.Unmarshal(payload)
+	if err != nil || len(vp9.Payload) < 1 {
+		return false
+	}
+	if !vp9.B {
+		return false
+	}
+
+	if (vp9.Payload[0] & 0xc0) != 0x80 {
+		return false
+	}
+
+	profile := (vp9.Payload[0] >> 4) & 0x3
+	if profile != 3 {
+		return (vp9.Payload[0] & 0xC) == 0
+	}
+	return (vp9.Payload[0] & 0x6) == 0
```

Comment:
Don't fully understand this. I cannot read the bit stream format document :-). But, this RFC seems to say that we do not need to dig beyond the VP9 Payload Descriptor to determine key frame. I may change this function based on that when I do my refactor. Guess, we really do not need to know if key frame or not with dependency descriptor. We will need that only when using simulcast.

https://datatracker.ietf.org/doc/html/draft-ietf-payload-vp9-16. The actual text
```
      A key picture is a picture whose base
      spatial layer frame is a key frame, and which thus completely
      resets the encoder state.  This packet will have its P bit equal
      to zero, SID or L bit (described below) equal to zero, and B bit
      (described below) equal to 1.
```
This is what I used in the refactor branch I am working on. I will compare with the above and check if they match.